### DISCUSSION TITLE
Coupling between structural elements (beam and shell) and 3D solid elements

### DISCUSSION THREAD

**Discussion by Kavan-Khaledi** [idaholab/moose#21734] [Q&A Modules: Solid mechanics] [answered]
Dear MOOSE team,
I have successfully used the PorousFlow module to simulate tunneling in fully saturated porous rocks. Now, I am trying to add support elements such as rock bolts (beam elements) and shotcrete lining (shell elements) to the model. However, the structural elements (beam and shell) have 3 additional degrees of freedom (bending moments). Is it at all possible to couple (tie) a beam or shell element to a 3D solid element? I would be very grateful for your help.

Best Regards,
Kavan

**Comment by hugary1995** [comment, marked as answer]
There is an `EqualValueEmbeddedConstraint` you might find useful. https://mooseframework.inl.gov/source/constraints/EqualValueEmbeddedConstraint.html

**Reply by Kavan-Khaledi** [reply]
Thank you very much @hugary1995 :) The displacements of a solid element can be easily coupled to the displacements of a Beam element with this EmbeddedConstraint. Thanks;)
